API/RestMatch.API/RestMatch.API/Controllers/RestaurantsController.cs

```diff
@@ -0,0 +1,191 @@
+    [ApiController]
+    public class RestaurantsController : ControllerBase
+    {
+        private readonly IRestaurantService _service;
+
+        public RestaurantsController(IRestaurantService service)
+        {
+            _service = service;
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public async Task<ActionResult<ICollection<GetRestaurantResponseDto>>> GetRestaurants()
+        {
+            var restaurants = await _service.GetRestaurants();
+            if (restaurants.Count == 0)
+                return NoContent();
+
+            return Ok(restaurants);
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<GetRestaurantResponseDto>> GetRestaurant(int id)
+        {
+            var restaurant = await _service.GetRestaurant(id);
+            if (restaurant == null)
+                return NotFound($"{nameof(Restaurant)} wasn't found");
+
+            return Ok(restaurant);
+        }
+
+        [HttpGet("imageUrls/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<GetRestaurantImageUrlResponseDto>> GetRestaurantImageUrl(int id)
+        {
+            var imageUrl = await _service.GetRestaurantImageUrl(id);
+            if (imageUrl == null)
+                return NotFound($"{nameof(RestaurantImageUrl)} wasn't found");
+
+            return Ok(imageUrl);
+        }
+
+        [HttpGet("{restaurantId}/imageUrls")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<ICollection<GetRestaurantImageUrlResponseDto>>> GetRestaurantImageUrls(int restaurantId)
+        {
+            var imageUrls = await _service.GetRestaurantImageUrls(restaurantId);
+            if (imageUrls == null)
+                return NotFound($"{nameof(Restaurant)} wasn't found");
+
+            if (imageUrls.Count == 0)
+                return NoContent();
+
+            return Ok(imageUrls);
+        }
+
+        [HttpGet("cuisines/{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<GetRestaurantCuisineResponseDto>> GetRestaurantCuisine(int id)
+        {
+            var cuisine = await _service.GetRestaurantCuisine(id);
+            if (cuisine == null)
+                return NotFound($"{nameof(RestaurantCuisine)} wasn't found");
+
+            return Ok(cuisine);
+        }
+
+        [HttpGet("{restaurantId}/cuisines")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
```

Comment:
add also type of response for nswag documentation like [ProducesResponseType(typeof(ICollection<GetRestaurantCuisineResponseDto>), StatusCodes.Status200OK)]

Reply:
added